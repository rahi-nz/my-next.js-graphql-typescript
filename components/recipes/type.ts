export interface Props {
  photo: string;
  title: string;
  id: string;
}
