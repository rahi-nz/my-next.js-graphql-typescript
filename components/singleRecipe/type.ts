import { gql } from "apollo-boost";

export const RECIPE = gql`
  query($id: String!) {
    recipe(id: $id) {
      chef {
        name
      }
      description
      title
      photo {
        url
      }
      tagsCollection {
        items {
          name
        }
      }
    }
  }
`;

export interface Props {
  id: string;
}

interface chef {
  name: string | null;
}

interface photo {
  url: string;
}

interface tag {
  name: string;
}

interface tagsCollection {
  items: Array<tag>;
}

interface recipe {
  chef: chef;
  description: string;
  title: string;
  photo: photo;
  tagsCollection: tagsCollection;
}

export interface getRecipes {
  recipe: recipe;
}
