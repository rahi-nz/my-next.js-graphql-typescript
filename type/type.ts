import { gql } from "apollo-boost";

export const RECIPES = gql`
  query {
    recipeCollection {
      items {
        sys {
          id
        }
        title
        photo {
          url
        }
      }
    }
  }
`;

interface items {
  id: string;
}

interface photo {
  url: string;
}

interface items {
  sys: items;
  title: string;
  photo: photo;
}

interface Collection {
  items: Array<items>;
}

export interface recipesData {
  recipeCollection: Collection;
}
