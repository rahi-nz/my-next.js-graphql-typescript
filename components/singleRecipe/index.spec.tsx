import React from "react";
import renderer from "react-test-renderer";
import wait from "waait";
import { MockedProvider } from "@apollo/client/testing";
import { Props, RECIPE } from "./type";
import SingleRecipe from "./index";

describe("<SingleRecipe/>", () => {
  let testRenderer;

  const props: Props = {
    id: "4dT8tcb6ukGSIg2YyuGEOm",
  };

  it("should render singleRecipe", async () => {
    const recipeMock = {
      request: { query: RECIPE, variables: { id: "4dT8tcb6ukGSIg2YyuGEOm" } },
      result: {
        data: {
          recipe: {
            chef: null,
            description:
              "*Grilled Cheese 101*: Use delicious cheese and good quality bread; make crunchy on the outside and ooey gooey on the inside; add one or two ingredients for a flavor punch! In this case, cherry preserves serve as a sweet contrast to cheddar cheese, and basil adds a light, refreshing note. Use __mayonnaise__ on the outside of the bread to achieve the ultimate, crispy, golden-brown __grilled cheese__. Cook, relax, and enjoy!",
            title: "White Cheddar Grilled Cheese with Cherry Preserves & Basil",
            photo: {
              url:
                "https://images.ctfassets.net/kk2bw5ojx476/61XHcqOBFYAYCGsKugoMYK/0009ec560684b37f7f7abadd66680179/SKU1240_hero-374f8cece3c71f5fcdc939039e00fb96.jpg",
            },
            tagsCollection: {
              items: [
                {
                  name: "vegan",
                },
              ],
            },
          },
        },
      },
    };

    testRenderer = renderer.create(
      <MockedProvider mocks={[recipeMock]} addTypename={false}>
        <SingleRecipe {...props} />
      </MockedProvider>
    );
    await wait(0); // wait for response

    expect(
      testRenderer.root.findAllByProps({ "data-test-id": "title" }).length
    ).toBe(1);
    expect(
      testRenderer.root.findAllByProps({ "data-test-id": "TagContainer" })
        .length
    ).toBe(1);

    expect(
      testRenderer.root.findAllByProps({ "data-test-id": "title" })[0].props
        .children
    ).toBe("White Cheddar Grilled Cheese with Cherry Preserves & Basil");
  });

  it("Should render correctly", () => {
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
