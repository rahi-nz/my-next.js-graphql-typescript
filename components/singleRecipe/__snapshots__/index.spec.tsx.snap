// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleRecipe/> Should render correctly 1`] = `
<div
  className="style__Container-sc-1xyuyr6-0 gWhydR"
>
  <p>
    Chef: 
    Unknown
  </p>
  <img
    alt="White Cheddar Grilled Cheese with Cherry Preserves & Basil"
    className="style__Photo-sc-1xyuyr6-2 liSHuS"
    src="https://images.ctfassets.net/kk2bw5ojx476/61XHcqOBFYAYCGsKugoMYK/0009ec560684b37f7f7abadd66680179/SKU1240_hero-374f8cece3c71f5fcdc939039e00fb96.jpg"
  />
  <p
    data-test-id="title"
  >
    White Cheddar Grilled Cheese with Cherry Preserves & Basil
  </p>
  <div
    className="style__TagContainer-sc-1xyuyr6-1 fnbgGQ"
    data-test-id="TagContainer"
  >
    <span>
      vegan
    </span>
  </div>
  <div
    className="style__Description-sc-1xyuyr6-3 cKImFA"
  >
    <p>
      What's cooking
    </p>
    *Grilled Cheese 101*: Use delicious cheese and good quality bread; make crunchy on the outside and ooey gooey on the inside; add one or two ingredients for a flavor punch! In this case, cherry preserves serve as a sweet contrast to cheddar cheese, and basil adds a light, refreshing note. Use __mayonnaise__ on the outside of the bread to achieve the ultimate, crispy, golden-brown __grilled cheese__. Cook, relax, and enjoy!
  </div>
</div>
`;
